internal/character/arlan/eidolon.go

```diff
@@ -0,0 +1,102 @@
+package arlan
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	E1 key.Modifier = "arlan-e1"
+	E4 key.Modifier = "arlan-e4"
+	E6 key.Modifier = "arlan-e6"
+)
+
+func init() {
+	// When HP is lower than or equal to 50% of Max HP, increases Skill's DMG by 10%.
+	modifier.Register(E1, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHit: func(mod *modifier.ModifierInstance, e event.HitStartEvent) {
+				if e.Hit.AttackType != model.AttackType_SKILL {
+					return
+				}
+
+				if mod.Engine().HPRatio(mod.Owner()) <= 0.5 {
+					e.Hit.Attacker.AddProperty(prop.AllDamagePercent, 0.1)
+				}
+			},
+		},
+	})
+
+	// When struck by a killing blow after entering battle, instead of becoming knocked down,
+	// Arlan immediately restores his HP to 25% of his Max HP. This effect is automatically
+	// removed after it is triggered once or after 2 turn(s) have elapsed.
+	modifier.Register(E4, modifier.Config{
+		Duration:   2,
+		StatusType: model.StatusType_STATUS_BUFF,
+		Listeners: modifier.Listeners{
+			OnLimboWaitHeal: func(mod *modifier.ModifierInstance) bool {
+				mod.Engine().Heal(info.Heal{
+					Targets:  []key.TargetID{mod.Owner()},
+					Source:   mod.Owner(),
+					BaseHeal: info.HealMap{model.HealFormula_BY_TARGET_MAX_HP: 0.25},
+				})
+				mod.Engine().RemoveModifier(mod.Owner(), E4)
+				return true
+			},
```

Comment:
1. Before the heal, Arlan's revive immediately dispels all debuffs currently on him (call `DispelStatus` w/o specifying count and set order to `LastAdded`)
2. This heal needs to be deferred to the "revive stage" of combat. To do this you need to do `mod.Engine().InsertAbility(...)`. The priority of this insert should be `CharReviveSelf`
3. Based on the datamine, this is not a Heal but a `SetHP` call (so wont be buffed by any healing bonuses, just does `MaxHP * 0.25`)
4. NIT: `mod.RemoveSelf()` is a bit cleaner